core/common/src/main/java/alluxio/PropertyKey.java

```diff
@@ -11,378 +11,593 @@
 
 package alluxio;
 
+import com.google.common.base.Objects;
+
+import java.util.Collection;
 import java.util.HashMap;
 import java.util.Map;
 
 import javax.annotation.concurrent.ThreadSafe;
 
 /**
- * Configurations properties constants. Please check and update Configuration-Settings.md file when
- * you change or add Alluxio configuration properties.
+ * Configuration property key which is essentially a wrapper of String.
```

Comment:
I don't think that "which is essentially a wrapper of String" is accurate.

I would simply say "This class is used for representing configuration property keys."